documentation/system-design/book-bazar-openapi.yaml

```diff
@@ -0,0 +1,609 @@
+openapi: 3.0.1
+info:
+  title: Book Bazar
+  description: "[Book Bazar](https://bookbazar.me) API documentation."
+  version: 0.0.0
+externalDocs:
+  description: Check out Book Bazar on GitHub
+  url: https://github.com/calebmech/Book-Bazar
+servers:
+  - url: https://www.bookbazar.me/api
+  - url: https://localhost:4000/api
+tags:
+  - name: Search
+  - name: Post
+  - name: Book
+  - name: Course
+  - name: Auth
+  - name: User
+paths:
+  /search:
+    get:
+      tags:
+        - Search
+      summary: Search courses and books
+      parameters:
+        - name: q
+          in: query
+          description: Search query
+          required: true
+          schema:
+            type: string
+        - name: coursesLength
+          in: query
+          description: Maximum number of courses to return
+          schema:
+            type: integer
+            minimum: 0
+            maximum: 20
+            default: 3
+        - name: booksLength
```

Comment:
This shouldn't conflict since we'll be accessing it as an object property. I've removed this API. If we end up wanting to change it for the other APIs we can do this later in favour of getting this PR merged.